edgedb/datatypes/range.py

```diff
@@ -78,22 +78,31 @@ def is_empty(self) -> bool:
     def __bool__(self):
         return not self.is_empty()
 
-    def __eq__(self, other):
-        if not isinstance(other, Range):
+    def __eq__(self, other) -> bool:
```

Comment:
I don't think we should have try to have `Range` and `MultiRange` able to compare equal. 
In particular, if we are going to do that, we need to have `__hash__` be equal when the objects are equal, which is not worth dealing with.

Reply:
Good point about hashes equality. I'll drop the equality between Range and MultiRange.